src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -421,23 +421,41 @@ static ASTPtr getOrderByPolicy(
 
 void InterpreterCreateImpl::validate(const InterpreterCreateImpl::TQuery & create_query, ContextPtr)
 {
-    /// This is dangerous, because the like table may not exists in ClickHouse
-    if (create_query.like_table)
-        throw Exception("Cannot convert create like statement to ClickHouse SQL", ErrorCodes::NOT_IMPLEMENTED);
-
-    const auto & create_defines = create_query.columns_list->as<MySQLParser::ASTCreateDefines>();
-
-    if (!create_defines || !create_defines->columns || create_defines->columns->children.empty())
-        throw Exception("Missing definition of columns.", ErrorCodes::EMPTY_LIST_OF_COLUMNS_PASSED);
+    if (!create_query.like_table)
+    {
+        bool missing_columns_definition = true;
+        if (create_query.columns_list)
+        {
+            const auto & create_defines = create_query.columns_list->as<MySQLParser::ASTCreateDefines>();
+            if (create_defines && create_defines->columns && !create_defines->columns->children.empty())
+                missing_columns_definition = false;
+        }
+        if (missing_columns_definition)
+            throw Exception("Missing definition of columns.", ErrorCodes::EMPTY_LIST_OF_COLUMNS_PASSED);
+    }
 }
 
 ASTs InterpreterCreateImpl::getRewrittenQueries(
     const TQuery & create_query, ContextPtr context, const String & mapped_to_database, const String & mysql_database)
 {
-    auto rewritten_query = std::make_shared<ASTCreateQuery>();
+    if (create_query.like_table)
+    {
+        auto * table_like = create_query.like_table->as<ASTTableIdentifier>();
+        String table_name = table_like->shortName();
+        String db_name = table_like->compound() ? table_like->getTableId().database_name : mapped_to_database;
```

Comment:
Should we throw an exception if `table_like` is from another database?

Reply:
If this throws an exception, replication will stop, and the only way to resume it is to detach the database, manually edit the .metadata file to skip the event, and attach. If we ignore, the table will not be created and row changes will be ignored.

Neither is ideal (a "proper" fix could involve actually grabbing the schema of that other database, but that's definitely a different PR, and not something we have looked into). Having to choose between the two options I'd argue having a missing table is better than not being able to resume replication.

I fixed the logic for this a bit to reflect this properly, pushing that shortly.